### PR TITLE
Eligible hedvig forever (again)

### PR DESCRIPTION
**Pull request by FredrikAreschoug**
The logic here is if a member has redeemed referral campaign or a redeemed hedvig forever campaign they are eligible for referrals

[Notion ticket](https://www.notion.so/hedviginsurance/71cf10addf9e42a2ae4e3e11d1d47a5d?v=698b1693c73d488996e38a2f36af5e58&p=7909f763d4fb4a2198d6e77049cc1f1e)

### CODE REVIEW

File: src/main/java/com/hedvig/productPricing/campaign/service/DomainService.kt [hunk 1]
```diff
@@ -319,19 +319,18 @@ class DomainService(
...
     val referralCampaign = campaign as ReferralCampaign
     val referrerMemberId = (referralCampaign.campaignOwner as Member).memberId
 
-    val referrerMarketFromContract = contractService.getMarketFromContractState(referrerMemberId)
+    val referrerMarketFromContract = contractService.getContractMarketInfoByHolderMemberId(referrerMemberId)?.market
       ?: throw NullPointerException("CampaignOwner with memberId ${referrerMemberId} has no market")
 
     val referentMarket =
-      contractService.getMarketFromContractState(referentMemberId)
+      contractService.getContractMarketInfoByHolderMemberId(referentMemberId)?.market
```
Comment (vonElfvin) [inline src/main/java/com/hedvig/productPricing/campaign/service/DomainService.kt:330]: Can a member have a contract if they are referred?

Comment (FredrikAreschoug) [reply src/main/java/com/hedvig/productPricing/campaign/service/DomainService.kt:330]: yes

File: src/main/java/com/hedvig/productPricing/campaign/service/exception/NoHedvigForeverCampaignException.kt [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package com.hedvig.productPricing.campaign.service.exception
+
+import com.hedvig.contract.shared.models.objects.Market
+
+class NoHedvigForeverCampaignException(market: Market) : RuntimeException("No hedvig forever campaign for $market")
```
Comment (vonElfvin) [inline src/main/java/com/hedvig/productPricing/campaign/service/exception/NoHedvigForeverCampaignException.kt:5]: Super subjective but I personally dislike having marketing words in class names etc. the concept and the market name should be spearated IMO. 
```suggestion
class NoForeverCampaignException(market: Market) : RuntimeException("No hedvig forever campaign for $market")
```

File: src/main/java/com/hedvig/productPricing/campaign/web/MemberCampaignController.kt [hunk 3]
```diff
@@ -82,4 +88,16 @@ class MemberCampaignController(
...
   }
+
+  @PostMapping("{market}/manualRedeemHedvigForever")
+  fun manualRedeemHedvigForever(
+    @PathVariable market: Market,
+    @Valid @RequestBody request: ManualRedeemHedvigForeverRequest
+  ): ResponseEntity<ManualRedeemHedvigForeverResponse> =
+    ResponseEntity.ok(
+      ManualRedeemHedvigForeverResponse(
+        redeemCampaignService.redeemHedvigForever(market, request.memberIds)
```
Comment (vonElfvin) [inline src/main/java/com/hedvig/productPricing/campaign/web/MemberCampaignController.kt:99]: Does this mean you can get 10 SEK discount without being invited or inviting?

**Comment by FredrikAreschoug** [comment]
@vonElfvin If you read the discription of the or I would want you to review #566 before reviewing this one. That's why this one is a draft

File: src/main/java/com/hedvig/productPricing/campaign/web/MemberCampaignController.kt (hunk 3 above)
Comment (FredrikAreschoug) [reply src/main/java/com/hedvig/productPricing/campaign/web/MemberCampaignController.kt:99]: yes but it's up to us to set the incentive on the campaign

File: src/main/java/com/hedvig/productPricing/campaign/service/exception/NoHedvigForeverCampaignException.kt (hunk 2 above)
Comment (FredrikAreschoug) [reply src/main/java/com/hedvig/productPricing/campaign/service/exception/NoHedvigForeverCampaignException.kt:5]: In my opinion, `ForeverCampaign` becomes something completely different. So either go with `HedvigForever` or rename the entire feature to `EnableReferralsCampaign`. I like going with `HedvigForever`

**Pull request merged**